xen/include/public/io/sndif.h

```diff
@@ -246,17 +307,77 @@
 #define XENSND_OP_WRITE                 3
 #define XENSND_OP_SET_VOLUME            4
 #define XENSND_OP_GET_VOLUME            5
+#define XENSND_OP_MUTE                  6
+#define XENSND_OP_UNMUTE                7
 
 /*
  * The maximum amount of shared pages which can be used in any request
  * from the frontend driver to the backend driver
+ * Pages must be of XC_PAGE_SIZE length.
  */
 #define XENSND_MAX_PAGES_PER_REQUEST    10
```

Comment:
What if there will be 128-channel stream with F64 format? One sample will have size of 1024 bytes. 4 samples per standard page, 40 samples at all. 
Or consider more realistic Hi-Fi device: 8 32-bit channels, sample rate 192kHz. Ten pages of 4096 bytes will hold 0.006 seconds of audio. Are you sure, that you will be able to submit new request 167 times per second?